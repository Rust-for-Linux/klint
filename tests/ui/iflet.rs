#![crate_type = "lib"]

// This is a case that currently klint will return false positive.

pub struct X;

impl Drop for X {
    #[klint::preempt_count(expect = 0)]
    #[inline(never)]
    fn drop(&mut self) {}
}

#[klint::preempt_count(expect = 0..)]
pub fn foo(x: Option<X>) -> Option<X> {
    if let Some(x) = x {
        return Some(x);
    }
    None
}
