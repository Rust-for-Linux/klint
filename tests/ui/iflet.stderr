WARN klint::ctxt no klint metadata found for crate core
error: function annotated to have preemption count expectation of 0..
  --> $DIR/iflet.rs:14:1
   |
14 | pub fn foo(x: Option<X>) -> Option<X> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: but the expectation inferred is 0
note: which may drop type `std::option::Option<X>` with preemption count 0..
  --> $DIR/iflet.rs:19:1
   |
14 | pub fn foo(x: Option<X>) -> Option<X> {
   |            - value being dropped is here
...
19 | }
   | ^
   = note: which may drop type `X` with preemption count 0..
note: which may call this function with preemption count 0..
  --> $DIR/iflet.rs:10:5
   |
10 |     fn drop(&mut self) {}
   |     ^^^^^^^^^^^^^^^^^^
   = note: but the callee expects preemption count 0

error: aborting due to 1 previous error

